datalad/support/gitrepo.py

```diff
@@ -495,13 +495,22 @@ def clone(self, url, path):
         path : str
         """
 
+        if is_ssh(url):
+            cnct = ssh_manager.get_connection(url)
+            cnct.open()
+            # TODO: with git <= 2.3 keep old mechanism:
+            #       with rm.repo.git.custom_environment(GIT_SSH="wrapper_script"):
+            env = {'GIT_SSH_COMMAND': "ssh -S %s" % cnct.ctrl_path}
+        else:
+            env = None
         ntries = 5  # 3 is not enough for robust workaround
         for trial in range(ntries):
             try:
                 lgr.debug("Git clone from {0} to {1}".format(url, path))
                 self.repo = self.cmd_call_wrapper(gitpy.Repo.clone_from,
                                                   url,
                                                   path,
+                                                  env=env,
```

Comment:
I think this is a problem. Not here but in the runner implementation. This will cause the entire process environment to be replaced with the above minimal environment definition. That might lead to unexpected things. One solution would be to copy os.environ, ammend it with the above setting, and pass it on.